src/main/java/com/icthh/xm/uaa/security/oauth2/idp/jwk/JwkAttributes.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.icthh.xm.uaa.security.oauth2.idp.jwk;
+
+import com.icthh.xm.uaa.security.oauth2.idp.converter.XmJwkVerifyingJwtAccessTokenConverter;
+import org.springframework.security.oauth2.provider.token.store.jwk.JwkTokenStore;
+
+/**
+ * Shared attribute values used by {@link JwkTokenStore} and associated collaborators.
+ * <p>
+ * This class copied from {@link org.springframework.security.oauth2.provider.token.store.jwk.JwkAttributes}
+ * and couldn't be imported cause of package private access.
+ * <p>
+ * Reason: we need custom implementation of JwkDefinitionSource class
+ * which impossible to import and override - it has package private access.
+ * <p>
+ * This class is required for {@link XmJwkVerifyingJwtAccessTokenConverter} implementation.
+ *
+ * @author Joe Grandja
+ * @author Michael Duergner
+ */
+public final class JwkAttributes {
```

Comment:
looks like library class, maybe better to add library to gradle that copy paste source code?